### ISSUE TITLE
icyMetadata not change 

### ISSUE THREAD

**Issue by FatihAydin52** [ryanheise/just_audio#102]
**Describe the bug**
When switching from one audio file to another, if there is an error in reading metadata, the metadata of the old audio file remains. 

**Minimal reproduction project**
https://github.com/FatihAydin52/just-audio-metadata

**To Reproduce**
1. Click play button. Now current song is None
3. Change Radio 1 Button
4. Click again Play button. Now current song is changed.
5. Change Radio 2 Button
6. Click again Play button. Now the current song has not changed.
7. Although the metadata for Radio 2 is empty, the metadata of the previous radio is retained.

**Error messages**
[+2771 ms] I/OMXClient( 6535): IOmx service obtained
[   +2 ms] I/ACodec  ( 6535): codec does not support config priority (err -2147483648)
[   +4 ms] I/ACodec  ( 6535): codec does not support config priority (err -2147483648)
[        ] W/ExtendedACodec( 6535): Failed to get extension for extradata parameter
```
```

**Expected behavior**
If there is an error reading the metadata, icyMetadata variable should return empty.

**Smartphone (please complete the following information):**
 - Device: Xioami Mi Note 3
 - OS: Android 9.0

**Flutter SDK version**
```
[√] Flutter (Channel stable, v1.17.2, on Microsoft Windows [Version 10.0.18363.836], locale tr-TR)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[√] Android Studio (version 4.0)
[√] Connected device (1 available)

• No issues found!
```

**Comment by ryanheise** [comment]
Hi @FatihAydin52 my apologies for taking so long to reply, I have had a long backlog of issues.

But I have added this to the list of issues I will look into next.

**Comment by FatihAydin52** [comment]
Okay. I am waiting. Thank you for your interest.

**Comment by ryanheise** [comment]
My sincerest apologies for again taking far too long to get onto this.

I have just tried your example, and I had to make a few changes to make it work with the latest version of just_audio code, but when I ran it, none of the included streams included icy metadata.

When I substituted in some different streams that did have icy metadata, it seemed to be working correctly in that after switching audio sources, the ICY metadata changed.

Let me know if the latest version works for you, and if there is still an issue, let me know if you have some other URLs in particular that you would like me to test.
